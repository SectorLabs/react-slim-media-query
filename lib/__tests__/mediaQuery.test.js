import React from 'react';

import renderer from 'react-test-renderer';
import MediaQuery from '../mediaQuery';

const createMockMediaMatcher = matches => () => ({
    matches,
    addListener: () => {},
    removeListener: () => {},
});

const runBasicTests = () =>
    [
        { minWidth: 1000 },
        { maxWidth: 1000 },
        { minDeviceWidth: 1000 },
        { maxDeviceWidth: 1000 },
        { minWidth: 1024, maxWidth: 1000 },
        { maxHeight: 1000 },
        { minHeight: 1000 },
        { minWidth: '1000px' },
        { maxWidth: '1000px', minWidth: 1024 },
        { query: 'max-device-width: 1000px' },
    ].forEach(values =>
        test('MediaQuery that matches using values', () => {
            const component = renderer.create(
                <MediaQuery {...values}>
                    <div>I should be heree</div>
                </MediaQuery>,
            );

            const tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        }),
    );

describe('Media query that match', () => {
    beforeEach(() => {
        window.matchMedia = createMockMediaMatcher(true);
    });

    runBasicTests();
});

describe("Media query that doesn't match", () => {
    beforeEach(() => {
        window.matchMedia = createMockMediaMatcher(false);
    });

    runBasicTests();
});
