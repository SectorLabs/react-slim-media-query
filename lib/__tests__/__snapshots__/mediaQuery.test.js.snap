// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media query that doesn't match MediaQuery that matches using values 1`] = `null`;

exports[`Media query that doesn't match MediaQuery that matches using values 2`] = `null`;

exports[`Media query that doesn't match MediaQuery that matches using values 3`] = `null`;

exports[`Media query that doesn't match MediaQuery that matches using values 4`] = `null`;

exports[`Media query that doesn't match MediaQuery that matches using values 5`] = `null`;

exports[`Media query that doesn't match MediaQuery that matches using values 6`] = `null`;

exports[`Media query that doesn't match MediaQuery that matches using values 7`] = `null`;

exports[`Media query that doesn't match MediaQuery that matches using values 8`] = `null`;

exports[`Media query that doesn't match MediaQuery that matches using values 9`] = `null`;

exports[`Media query that doesn't match MediaQuery that matches using values 10`] = `null`;

exports[`Media query that match MediaQuery that matches using values 1`] = `
<div>
  I should be heree
</div>
`;

exports[`Media query that match MediaQuery that matches using values 2`] = `
<div>
  I should be heree
</div>
`;

exports[`Media query that match MediaQuery that matches using values 3`] = `
<div>
  I should be heree
</div>
`;

exports[`Media query that match MediaQuery that matches using values 4`] = `
<div>
  I should be heree
</div>
`;

exports[`Media query that match MediaQuery that matches using values 5`] = `
<div>
  I should be heree
</div>
`;

exports[`Media query that match MediaQuery that matches using values 6`] = `
<div>
  I should be heree
</div>
`;

exports[`Media query that match MediaQuery that matches using values 7`] = `
<div>
  I should be heree
</div>
`;

exports[`Media query that match MediaQuery that matches using values 8`] = `
<div>
  I should be heree
</div>
`;

exports[`Media query that match MediaQuery that matches using values 9`] = `
<div>
  I should be heree
</div>
`;

exports[`Media query that match MediaQuery that matches using values 10`] = `
<div>
  I should be heree
</div>
`;
